packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -1,292 +1,279 @@
-  };
-}
+    return super.createCreationAction(
+      creationParameters,
+    ) as ExtensionTypes.IAction<TCreationParameters>;
+  }
 
-/**
- * Creates the extensionsData to add a refund address
- *
- * @param extensions extensions parameters to create
- *
- * @returns IAction the extensionsData to be stored in the request
- */
-function createAddRefundAddressAction(
-  extensionId: ExtensionTypes.ID,
-  addRefundAddressParameters: ExtensionTypes.PnAddressBased.IAddRefundAddressParameters,
-): ExtensionTypes.IAction {
-  return {
-    action: ExtensionTypes.PnAddressBased.ACTION.ADD_REFUND_ADDRESS,
-    id: extensionId,
-    parameters: {
-      refundAddress: addRefundAddressParameters.refundAddress,
-    },
-  };
-}
+  /**
+   * Creates the extensionsData to add a payment address
+   *
+   * @param extensions extensions parameters to create
+   *
+   * @returns IAction the extensionsData to be stored in the request
+   */
+  public createAddPaymentAddressAction(
+    addPaymentAddressParameters: ExtensionTypes.PnAddressBased.IAddPaymentAddressParameters,
+  ): ExtensionTypes.IAction {
+    if (
+      addPaymentAddressParameters.paymentAddress &&
+      !this.isValidAddress(addPaymentAddressParameters.paymentAddress)
+    ) {
+      throw Error('paymentAddress is not a valid address');
+    }
 
-/**
- * Applies the extension action to the request
- * Is called to interpret the extensions data when applying the transaction
- *
- * @param isValidAddress address validator function
- * @param extensionsState previous state of the extensions
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- *
- * @returns state of the request updated
- */
-function applyActionToExtension(
-  isValidAddress: (address: string) => boolean,
-  extensionsState: RequestLogicTypes.IExtensionStates,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): RequestLogicTypes.IExtensionStates {
-  const copiedExtensionState: RequestLogicTypes.IExtensionStates = Utils.deepCopy(extensionsState);
+    return {
+      action: ExtensionTypes.PnAddressBased.ACTION.ADD_PAYMENT_ADDRESS,
+      id: this.extensionId,
+      parameters: {
+        paymentAddress: addPaymentAddressParameters.paymentAddress,
+      },
+    };
+  }
 
-  if (extensionAction.action === ExtensionTypes.PnAddressBased.ACTION.CREATE) {
-    if (requestState.extensions[extensionAction.id]) {
-      throw Error(`This extension has already been created`);
+  /**
+   * Creates the extensionsData to add a refund address
+   *
+   * @param extensions extensions parameters to create
+   *
+   * @returns IAction the extensionsData to be stored in the request
+   */
+  public createAddRefundAddressAction(
+    addRefundAddressParameters: ExtensionTypes.PnAddressBased.IAddRefundAddressParameters,
+  ): ExtensionTypes.IAction {
```

Comment:
Maybe for another PR / refacto but we should align type namings `ExtensionTypes.IAction` with comments "Creates the extensionsData". Is it an action or an extension data?